packages/foundry/contracts/hooks/OrderHook.sol

```diff
@@ -0,0 +1,476 @@
+    /// @param permit2 Contract to approve tokens
+    constructor(
+        IVault vault,
+        address allowedFactory,
+        address trustedRouter,
+        IPermit2 permit2
+    ) VaultGuard(vault) {
+        i_allowedFactory = allowedFactory;
+        i_trustedRouter = trustedRouter;
+        i_vault = vault;
+        i_permit2 = permit2;
+    }
+
+    // Receive function to receive ether
+    receive() external payable {}
+
+    // Fallback function to receive ether
+    fallback() external payable {}
+
+    /// @notice Get the hook flags
+    function getHookFlags()
+        public
+        pure
+        override
+        returns (HookFlags memory hookFlags)
+    {
+        hookFlags.shouldCallAfterSwap = true;
+    }
+
+    /// @notice Check if the pool is allowed to register with this hook
+    /// @dev This hook implements a restrictive approach, where we check if the factory is an allowed factory and if the pool is created by the allowed factory
+    /// @param factory The factory contract that is deploying the pool
+    /// @param pool The pool that is being registered with this hook
+    function onRegister(
+        address factory,
+        address pool,
+        TokenConfig[] memory /* tokenConfig */,
+        LiquidityManagement calldata /* liquidityManagement */
+    ) public view override onlyVault returns (bool) {
+        // This hook implements a restrictive approach, where we check if the factory is an allowed factory and if
+        // the pool was created by the allowed factory
+        return
+            factory == i_allowedFactory &&
+            IBasePoolFactory(factory).isPoolFromFactory(pool);
+    }
+
+    /*///////////////////////////////////
+    ///////////    HOOK       ///////////
+    ///////////////////////////////////*/
+
+    /// @notice After swap hook
+    /// @dev This hook is called after a swap is executed. We store the previous prices of the tokens in the pool
+    /// @dev This emits an event with the current prices so that we can use that offchain to evalaute the positions of the current active orders
+    /// @dev Only the vault and trusted router is allowed to call this hook
+    /// @param params The parameters of the swap
+    function onAfterSwap(
+        AfterSwapParams calldata params
+    )
+        public
+        override
+        onlyVault
+        returns (bool success, uint256 hookAdjustedAmountCalculatedRaw)
+    {
+        if (params.router != i_trustedRouter) {
+            revert OrderHook__UnauthorizedRouter();
+        }
+
+        emit AfterSwapPrice(
+            params.pool,
+            address(params.tokenIn),
+            address(params.tokenOut),
+            params.tokenInBalanceScaled18,
+            params.tokenOutBalanceScaled18
+        );
+
+        prevPrices[params.pool][params.tokenIn] = params.tokenInBalanceScaled18;
+        prevPrices[params.pool][params.tokenOut] = params
+            .tokenOutBalanceScaled18;
```

Comment:
Why are token balances stored as "prices"?

Prices depend on the token balances, yes, but also on the pool invariant. Not sure whether I follow the logic here.

Reply:
Also, this is not used anywhere else. Is this supposed to be used offchain?

Reply:
Thank you for catching that. It seems I initially intended to use these stored balances for managing and executing orders but later changed the architecture, so this code is now redundant. I’ll remove it to streamline the contract and avoid any confusion about its purpose.